qiskit/aqua/algorithms/phase_estimators/phase_estimator.py

```diff
@@ -0,0 +1,213 @@
+            idx = np.argmax(abs(self._phases)) # np.argmax ignores complex part of number. But, we take abs anyway
+            binary_phase_string = np.binary_repr(idx, self._num_evaluation_qubits)[::-1]
+
+        else:
+            binary_phase_string = max(self._phases, key=self._phases.get)
+
+        phase = bit_string_to_phase(binary_phase_string)
+        return phase
+
+    def _run(self):
+        """
+        Run the circuit and return the estimated phase as a number between 0.0 and 1.0, with 1.0 corresponding
+        to a phase of 2pi.
+        """
+
+        result = self._quantum_instance.execute(self._pe_circuit)
+        self._result = result
+        self._compute_phases()
+
+    def filter_phases(self, cutoff, as_float=False):
+        """
+        Return a dict whose keys are phases and values are frequencies (counts)
+        keeping only frequencies (counts) larger than cutoff. It is assumed that
+        the `run` method has been called so that the phases have been computed.
+        When using a noiseless, shot-based simulator to read a single phase that can
+        be represented exactly by `num_evaluation_qubits`, all the weight will
+        be concentrated on a single phase. In all other cases, many, or all, bit
+        strings will have non-zero weight. This method is useful for filtering
+        out these uninteresting bit strings.
+
+        Args:
+            cutoff: minimum weight of number of counts required to keep a bit string.
+            as_float: If `True`, returned keys are floats in `[0.0, 1)`. If `False`
+                      returned keys are bit strings.
+        """
+        if isinstance(self._phases, qiskit.result.Counts):
+            counts = self._phases
+            if as_float:
+                phases = {bit_string_to_phase(k) : counts[k] for k in counts.keys() if counts[k] > cutoff}
+            else:
+                phases = {k : counts[k] for k in counts.keys() if counts[k] > cutoff}
+
+        else:
+            phases = {}
+            if as_float:
+                for idx, amplitude in enumerate(self._phases):
+                    if amplitude > cutoff:
+                        binary_phase_string = np.binary_repr(idx, self._num_evaluation_qubits)[::-1]
+                        phases[bit_string_to_phase(binary_phase_string)] = amplitude
+            else:
+                for idx, amplitude in enumerate(self._phases):
+                    if amplitude > cutoff:
+                        binary_phase_string = np.binary_repr(idx, self._num_evaluation_qubits)[::-1]
+                        phases[binary_phase_string] = amplitude
+
+            phases = _sort_phases(phases)
+
+        return phases
+
+    def phases_as_float(self):
+        """
+        Return a dictionary whose keys are phases as floats in `[0, 1)` and values are
+        freqencies or counts.
+        """
+        return self.filter_phases(0, as_float=True)
+
+
+def bit_string_to_phase(binary_string):
```

Comment:
Not sure if this qualifies to be a user-facing method, it seems quite straightforward. But if you think its useful and used in other places (e.g. your tutorial) then I'm fine with it

Reply:
I've been thinking the same thing. Not clear. I planned to make it internal-only if it seems best.